source/common/filter/config_discovery_impl.h

```diff
@@ -176,8 +183,40 @@ class HttpDynamicFilterConfigProviderImpl
   Http::FilterFactoryCb instantiateFilterFactory(const Protobuf::Message& message) const override {
     auto* factory = Registry::FactoryRegistry<Server::Configuration::NamedHttpFilterConfigFactory>::
         getFactoryByType(message.GetTypeName());
-    return factory->createFilterFactoryFromProto(message, getStatPrefix(), getFactoryContext());
+    return factory->createFilterFactoryFromProto(message, getStatPrefix(), factory_context_);
   }
+
+  Server::Configuration::FactoryContext& factory_context_;
+};
+
+// Implementation of a listener dynamic filter config provider.
+class ListenerDynamicFilterConfigProviderImpl
```

Comment:
A question for UDP, this seems a tcp config provider since the the factory cb is for tcp `Network::ListenerFilterFactoryCb`. What about UDP here? It is going to separate implement or this can be a base class for tcp and udp.

Reply:
Good catch. Added the support for UDP listener. PTAL.